src/dataset/scads_dataset.py

```diff
@@ -1,100 +1,80 @@
-                for e in entries
-                if e[45:49].lower() == s
-            ]
-            pairs = list(zip(entries[:-1], entries[1:]))
-            all_pairs.extend(pairs)
-    return all_pairs
-
-
-def _shuffle(elems, rnd):
-    elems = elems[:]
-    rnd.shuffle(elems)
-    return elems
-
-
-def read_kg(base_path: str, source: str, ids: Dict[str, int]) -> KG:
-    g = rdflib.graph.Graph()
-    g.parse(f"{base_path}/{source}_snippet.nt", format="nt")
-    return convert_rdf_to_openea(g, ids)
 
 
 class ScadsDataset(Dataset):
-    def __init__(
-        self,
-        base_path: str,
-        source1: str,
-        source2: str,
-        model: BasicModel,
-        rnd: random.Random,
-        train_size=0.7,
-        val_size=0.2,
-    ):
-        source1 = source1.lower()
-        source2 = source2.lower()
-        e_type_list = ["company", "episode", "movie", "person", "tvSeries"]
-        ids = {}
+    def __init__(self, data_folder: str, division: str, args_path: str):
+        args = load_args(args_path)
+        self._data_folder = data_folder
+        self.download_and_unzip()
+        self._kgs = read_kgs_from_folder(
+            data_folder, division, args.alignment_module, args.ordered
+        )
+        train_links = [(e[0], e[1], 1) for e in self._kgs.train_links]
+        valid_links = [(e[0], e[1], 1) for e in self._kgs.valid_links]
+        test_links = [(e[0], e[1], 1) for e in self._kgs.test_links]
+        super().__init__(
+            self._kgs.kg1,
+            self._kgs.kg2,
+            random.Random(),
+            train_links,
+            valid_links,
+            None,
+            test_links,
+        )
+        self._name = data_folder.split("/")[-2] + "/" + division[:-1]
 
-        kg1 = read_kg(base_path, source1, ids)
-        kg2 = read_kg(base_path, source2, ids)
-        pos_pairs_per_type = {
-            e_type: _shuffle(
-                _get_positive_pairs(base_path, e_type, [source1, source2]), rnd,
+    def download_and_unzip(self):
+        if not os.path.exists(self._data_folder):
+            parent_dir = self._get_download_dir()
+            print(
+                f"Did not find movie graph data in {self._data_folder}\nTherefore downloading and creating data"
+            )
+            repo_path = os.path.join("data", "MovieGraphBenchmark")
+            subprocess.check_call(
+                [
+                    "git",
+                    "clone",
+                    "--quiet",
+                    "https://github.com/ScaDS/MovieGraphBenchmark/",
+                    repo_path,
+                ]
             )
-            for e_type in e_type_list
-        }
+            # execute it as a script
+            subprocess.check_call(
+                [
+                    "python",
+                    os.path.join(
+                        "data", "MovieGraphBenchmark", "src", "create_graph.py"
+                    ),
+                ]
+            )
```

Comment:
This was the simplest way to make this work without having to mess with the pythonpath etc.